<?xml version="1.0"?>
<launch>

  <!-- send arm urdf to parameter server -->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find cob_defs)/ros/robots/cob3_torso.urdf.xacro'" />

  <!-- send parameters to parameter server -->
  <param name="CanDevice" value="15"/>
  <param name="CanBaudrate" value="500"/>
  <rosparam command="load" file="$(find cob_powercube_chain)/ros/launch/torso.yaml"/>

  <!-- start powercube_chain -->
  <node pkg="cob_powercube_chain" type="cob_powercube_chain" name="cob_torso" cwd="node" respawn="false" output="screen" />

  <!-- start robot_state_publisher -->
  <node pkg="robot_state_publisher" type="state_publisher" name="rob_st_pub" />

  <!-- set Powercubes to OperationMode position -->
  <param name="OperationMode" value="position"/>

</launch>
